pulsar-metadata/src/main/java/org/apache/pulsar/metadata/api/MetadataCache.java

```diff
@@ -118,6 +118,19 @@
      */
     CompletableFuture<Void> create(String path, T value);
 
+    /**
+     * Create or update an object in the metadata store.
+     * <p>
+     * This operation will make sure to keep the cache consistent.
+     *
+     * @param path
+     *            the path of the object in the metadata store
+     * @param value
+     *            the object to insert in metadata store
+     * @return a future to track the completion of the operation
+     */
+    CompletableFuture<Void> updateOrCreate(String path, T value);
```

Comment:
we have used `readModifyUpdateOrCreate` at few places for this usecase.

Reply:
Good point. the usage here is slightly different in that we're not caring for the existing value, but we can avoid adding a new method just for that.